x/orderbook/bucket_test.go

```diff
@@ -0,0 +1,138 @@
+package orderbook
+
+import (
+	"time"
+	"testing"
+
+	"github.com/iov-one/weave"
+	"github.com/iov-one/weave/coin"
+	"github.com/iov-one/weave/errors"
+	"github.com/iov-one/weave/orm"
+	"github.com/iov-one/weave/weavetest"
+)
+
+func TestMarketIDindexer(t *testing.T) {
+	cases := map[string]struct {
+		obj orm.Object
+		wantErr *errors.Error 
+	}{
+		"success": {
+			obj: orm.NewSimpleObj(nil, new(OrderBook)), 
+			wantErr: nil,
+		},
+		"not orderbook": {
+			obj: orm.NewSimpleObj(nil, new(Order)),
+			wantErr: errors.ErrState,
+		},
+	}
+
+	for testName, tc := range cases {
+		t.Run(testName, func(t *testing.T) {
+			if _, err := marketIDindexer(tc.obj); !tc.wantErr.Is(err) {
+				t.Fatalf("unexpected error: %+v", err)
+			}
+		})
+	}
+}
+
+func TestOpenOrderIndexer(t *testing.T) {
```

Comment:
Nice idea to test these indexers.
It would be good to check return values as well as error cases.

* That if the OrderState is not Open it returns nil, nil
* That if values are all properly set, you get an expected (manually calculated) index key back.

The second is basically a BuildOpenOrderIndex test, but fair enough to do it here

Reply:
Wrote an extra function for creating index values manually. Would be enough I guess